src/types.rs

```diff
@@ -198,6 +220,17 @@ impl From<BlsPublicKey> for PublicKeyOrIndex {
     }
 }
 
+impl From<String> for PublicKeyOrIndex {
+    fn from(s: String) -> Self {
+        match try_bytes_from_hex_str(&s) {
+            Ok(..) => {
+                Self::PublicKey(try_bytes_from_hex_str(&s).unwrap().as_slice().try_into().unwrap())
+            }
+            _ => Self::Index(s.parse::<usize>().unwrap()),
+        }
```

Comment:
no reason to call `try_bytes_from_hex_str` twice...

I also think it will help w/ the nesting of checking for either public key or index if we just check the index first

```suggestion
        match s.parse::<ValidatorIndex>() {
            Ok(index) => Self::Index(index),
            Err(err) => {
                match try_bytes_from_hex_str(&s) {
                    Ok(bytes) => {
                        match BlsPublicKey::try_from(&bytes) {
                            Ok(public_key) => Self::PublicKey(public_key),
                            Err(err) => Err(format!("could not parse public key from hex string: {err}")),
                        }
                    }
                    Err(hex_err) => Err(format!("data could not be parsed as `ValidatorIndex` or `BlsPublicKey`: {err}; {hex_err}")),
                }
            }
        }
        
        match try_bytes_from_hex_str(&s) {
            Ok(..) => {
                Self::PublicKey(try_bytes_from_hex_str(&s).unwrap().as_slice().try_into().unwrap())
            }
            _ => Self::Index(s.parse::<usize>().unwrap()),
        }
```